src/common/core/setcore.js

```diff
@@ -69,9 +78,9 @@ define(['common/util/assert'], function (ASSERT) {
                 relid = Math.floor(Math.random() * MAX_RELID);
             } while (existingRelIds.indexOf(relid) !== -1);
             return '' + relid;
-        };
+        }
 
-        var harmonizeMemberData = function (node, setName) {
+        function harmonizeMemberData(node, setName) {
```

Comment:
Looks good in general, just a minor note for the future.
This function has 12 variables, 7 levels of indentation and three levels of nested loops and it's a recursive function. Do consider refactoring out the inner most loop into a helper function (ideally outside of the function scope).